docs/utils/sessions.md

```diff
@@ -265,9 +271,11 @@ The `expires` argument to `createData` and `updateData` is the same `Date` at wh
 
 For purely cookie-based sessions (where the session data itself is stored in the session cookie with the browser, see [cookies][cookies]) you can use `createCookieSessionStorage()`.
 
-The main advantage of cookie session storage is that you don't need any additional backend services or databases to use it. It can also be beneficial in some load-balanced scenarios. However, cookie-based sessions may not exceed the browser's max-allowed cookie length (typically 4kb).
+The main advantage of cookie session storage is that you don't need any additional backend services or databases to use it. It can also be beneficial in some load-balanced scenarios.
 
-The downside is that you have to `commitSession` in almost every loader and action. If your loader or action changes the session at all, it must be committed. That means if you `session.flash` in an action, and then `session.get` in another, you must commit it for that flashed message to go away. With other session storage strategies you only have to commit it when it's created (the browser cookie doesn't need to change because it doesn't store the session data, just the key to find it elsewhere).
+However, cookie-based sessions may not exceed the browser's cookie size limit, typically 1024 bytes per cookie value (and 4 KB total for all cookies).
```

Comment:
I think the original 4k is accurate here.  It's 4k per _total_ cookie length (including name/attrs) and 1k per attribute value per [Chrome's docs](https://chromestatus.com/feature/4946713618939904) and the [spec](https://datatracker.ietf.org/doc/html/rfc6265#section-6.1) also states 4k but doesn't say anything about the attribute length limit.

I did a quick test and as expected this is the largest cookie I can set before it gets ignored:

```
"Set-Cookie": `test=${new Array(4092).fill("a").join("")}`,
```

<img width="933" alt="Screenshot 2023-08-25 at 10 05 46 AM" src="https://github.com/remix-run/remix/assets/1609022/5ffeac6c-75c5-4409-8c23-a88ef279f9c8">


Let's just leave it at 4k and not say anything about the 1k attribute limit.

The only "combined" cookie limits I've run into is ~15k in GCP before you start getting 429 errors.

Reply:
```suggestion
However, cookie-based sessions may not exceed the browser's cookie size limit - typically 4k bytes per cookie.
```

Reply:
Thank you. I've made this change too, with an extra detail I did not know before:

> However, cookie-based sessions may not exceed browser cookie size limits of 4k bytes. If your cookie size exceeds this limit, `commitSession()` will throw an error.

This is good to call out, since it clarifies that exceeding the limit will result in an app crash. I was happy to see this safeguard present in the library—better to log an app crash than to have a page rejected by a browser.

https://github.com/remix-run/remix/blame/025ecac479def75dad3db4869ea1b4fb8917df2e/packages/remix-server-runtime/sessions/cookieStorage.ts#L53-L58